lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -1,19 +1,40 @@
 """
-Tests for the Tool Data Meister modules.
+Tests for the Tool Meister modules.
 """
+import io
+import logging
 import pathlib
+import shutil
 import subprocess
+from http import HTTPStatus
+
 import pytest
+import responses
 
-from pbench.agent.tool_meister import ToolMeister, get_logger
+from pbench.agent.tool_meister import ToolMeister, ToolMeisterError, get_logger
 
-logger = get_logger("__logger__")
+tar_file = "test.tar.xz"
 
 
 @pytest.fixture()
-def mock_tar(monkeypatch):
+def mock_tar(monkeypatch, agent_setup):
     def fake_run(*args, **kwargs):
-        return subprocess.CompletedProcess(args, returncode=0, stdout=b"", stderr=None)
+        assert kwargs["cwd"] == agent_setup["tmp"].parent
+        assert kwargs["stdin"] is None
+        assert kwargs["stderr"] == subprocess.STDOUT
+        assert kwargs["stdout"] == subprocess.PIPE
+        c = subprocess.CompletedProcess(args, returncode=0, stdout=b"", stderr=None)
+        assert all(
+            x in c.args[0]
+            for x in [
+                "tar_path",
+                "--create",
+                "--xz",
+                "--force-local",
+                f"--file={tar_file}",
+            ]
+        )
```

Comment:
It's a bit odd to be doing this assertion against your constructed `subprocess.CompletedProcess` rather than against the original argument. Functionally it's equivalent, but it inevitably and unnecessarily raises a momentary question in the reader's mind whether the method you called (even though it happens to be just a constructor) might have changed things.

On the other hand ... my first inclination was to also complain that you were merely checking that all expected arguments are in the list without regard to the order. Since all but one are options, though, there's no specific required order so  this is more flexible than `assert c.args == [ ... ]` as I might have suggested. (Although I really do hate what `black` does with long comprehensions. ☹️ )

Reply:
> It's a bit odd to be doing this assertion against your constructed `subprocess.CompletedProcess` rather than against the original argument. Functionally it's equivalent, but it inevitably and unnecessarily raises a momentary question in the reader's mind whether the method you called (even though it happens to be just a constructor) might have changed things.

Yeah its the same thing, but I see how that might be confusing to readers thinking whether the method changes them. On the other hand I also wanted to make sure the arguments did not get modified unexpectedly by the subprocess function and all the original arguments passed to the subprocess are intact.